### PR TITLE
Draft : [logo/pass] Refactor RemoveDeadNodeWithQueryPass

### PR DESCRIPTION
**Pull request by Hanjin-Choi**
This refactor RemoveDeadNodeWithQuery Pass.

Related : https://github.com/Samsung/ONE/issues/13863

ONE-DCO-1.0-Signed-off-by: HanJin Choi hanjin4647@gmail.com

### CODE REVIEW

**Comment by Hanjin-Choi** [comment]
The build failed without any changes.

image

**Comment by hseok-oh** [comment]
@nnfw-bot test nncc-debug

**Comment by Hanjin-Choi** [comment]
Build and dredd_Test success with code change

image

File: compiler/circle2circle-dredd-recipe-test/test.lst [hunk 1]
```diff
@@ -141,3 +141,5 @@ Add(REGRESS_ONNX_Conv_BN_Relu6_001 PASS
 
 Add(REGRESS_ONNX_Mul_Mul_000 PASS
       convert_nchw_to_nhwc)
+
+Add(RemoveDeadNodeWithQueryPass_000 PASS substitute_pack_to_reshape)
```
Comment (icodo98) [inline compiler/circle2circle-dredd-recipe-test/test.lst:145]: Maybe you should change test title

**Comment by hseok-oh** [comment]
@nnfw-bot test nncc-release

File: compiler/circle2circle-dredd-recipe-test/test.lst (hunk 1 above)
Comment (Hanjin-Choi) [reply compiler/circle2circle-dredd-recipe-test/test.lst:145]: I’m not sure if this title is correct.

Comment (jinevening) [reply compiler/circle2circle-dredd-recipe-test/test.lst:145]: I think `REGRESS_Issue_13863` would be ok.

**Comment by jinevening** [comment]
@nnfw-bot test nncc-debug

**Comment by jinevening** [comment]
@nnfw-bot test nncc-release

**Comment by jinevening** [comment]
Code looks good to me. After all tests are passed, please post PRs module-by-module.

**Comment by Hanjin-Choi** [comment]
> Code looks good to me. After all tests are passed, please post PRs module-by-module.

Thanks a lot! I will change Test title and seperate PRs. :)

**Pull request closed without merging**